### PR TITLE
Ocd: Review and test string assignments

### PR DESCRIPTION
**Pull request by dg0yt**
Resolves GH-1248.

### CODE REVIEW

File: src/fileformats/ocd_types.h [hunk 1]
```diff
@@ -125,7 +125,7 @@ namespace Ocd
 		
 		Utf16PascalString& operator=(const QString& value)
 		{
-			memcpy(reinterpret_cast<void*>(data), reinterpret_cast<const void*>(value.utf16()), std::max(std::size_t(value.length())+1, 2*N));
+			std::memcpy(data, value.utf16(), 2*std::min(static_cast<std::size_t>(value.length())+1, N));
```
Comment (dg0yt) [inline src/fileformats/ocd_types.h]: Note: it is `min`, not `max`.

**Comment by dg0yt** [comment]
I wasn't yet able to test with OC*D Viewer.

**Comment by dg0yt** [comment]
Last push: Updated copyright, removed extra `;` (-Wpedantic).

**Comment by lpechacek** [comment]
Will review. Hoping to get to it by the end of week.

**Comment by dg0yt** [comment]
Thanks.

What I could have done differently from the start is omitting the ocd_t.h file and put everything into the ocd_t.cpp file (cf. symbol_t.cpp). But I leave it as is for now.

**Comment by dg0yt** [comment]
- Removed `ocd_t.h`
- Reviewed commit messages and source code comments.
- Fixed GH issue reference.

I didn't add explanations for "why things are done as they are at the moment" because these strings like other OCD stuff are simply based on the documentation of this external file format. Maybe the naming of the string types could have been done differently, but this is out of scope now.

I intentionally did not introduce a function for common test data, because there are significant differences anyway, with regard to length and encoding.

I intentionally didn't want to mix commits changing behaviour with commits moving code around.

**Comment by lpechacek** [comment]
Fine with me. Thanks for the quick turnaround!

**Pull request merged**